In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv(r"C:\Users\HP\Desktop\RealEstatePricePrediction\data\processed\cleaned_data.csv")

In [3]:
print(df.shape)
df.head()


(8917, 6)


,location,total_sqft,bath,price,bhk,price_per_sqft
0,Devarabeesana Halli,1672.0,3.0,150.0,3,8971.29
1,Devarabeesana Halli,1750.0,3.0,149.0,3,8514.29
2,Devarabeesana Halli,1750.0,3.0,150.0,3,8571.43
3,Devarachikkanahalli,1250.0,2.0,44.0,3,3520.00
4,Devarachikkanahalli,1250.0,2.0,40.0,2,3200.00


In [4]:
print("Total unique locations:", df["location"].nunique())
print("\nLocation counts:\n", df["location"].value_counts().tail(20))

Total unique locations: 752

Location counts:
 location
T.C PALYA                         1
Teachers Colony                   1
Tigalarpalya                      1
Udaya Nagar                       1
Ullal Uppanagar                   1
Upadhyaya Layout                  1
Upkar Layout                      1
Varthur Road,                     1
Veer Sandra                       1
Veersandra                        1
Venugopal Reddy Layout            1
Versova Layout                    1
Vijinapura                        1
Vimanapura                        1
Virupakshapura                    1
Vivek Nagar                       1
Whitefield,                       1
Yeshwanthpur Industrial Suburb    1
frazertown                        1
manyata park                      1
Name: count, dtype: int64


In [5]:
location_counts = df["location"].value_counts()
rare_locations = location_counts[location_counts <= 10].index
df["location"] = df["location"].apply(lambda x: "other" if x in rare_locations else x)

print("Unique locations after grouping:", df["location"].nunique())

Unique locations after grouping: 176


- 752 locations before grouping, many with just 1 house in that particular location
- 176 locations after grouping, all with atleast 11 houses

In [6]:
#one hot encoding for location
df = pd.get_dummies(df, columns=["location"])
print("Shape after encoding:", df.shape)
print(df.head())

Shape after encoding: (8917, 181)
   total_sqft  bath  price  bhk  price_per_sqft  \
0      1672.0   3.0  150.0    3         8971.29   
1      1750.0   3.0  149.0    3         8514.29   
2      1750.0   3.0  150.0    3         8571.43   
3      1250.0   2.0   44.0    3         3520.00   
4      1250.0   2.0   40.0    2         3200.00   

   location_ Devarachikkanahalli  location_1st Phase JP Nagar  \
0                          False                        False   
1                          False                        False   
2                          False                        False   
3                           True                        False   
4                           True                        False   

   location_5th Phase JP Nagar  location_6th Phase JP Nagar  \
0                        False                        False   
1                        False                        False   
2                        False                        False   
3               

In [7]:
df = df.astype({col: int for col in df.select_dtypes(include='bool').columns})
print(df.head())

   total_sqft  bath  price  bhk  price_per_sqft  \
0      1672.0   3.0  150.0    3         8971.29   
1      1750.0   3.0  149.0    3         8514.29   
2      1750.0   3.0  150.0    3         8571.43   
3      1250.0   2.0   44.0    3         3520.00   
4      1250.0   2.0   40.0    2         3200.00   

   location_ Devarachikkanahalli  location_1st Phase JP Nagar  \
0                              0                            0   
1                              0                            0   
2                              0                            0   
3                              1                            0   
4                              1                            0   

   location_5th Phase JP Nagar  location_6th Phase JP Nagar  \
0                            0                            0   
1                            0                            0   
2                            0                            0   
3                            0                    

In [8]:
df.head(8917)

,total_sqft,bath,price,bhk,price_per_sqft,location_ Devarachikkanahalli,location_1st Phase JP Nagar,location_5th Phase JP Nagar,location_6th Phase JP Nagar,location_7th Phase JP Nagar,...,location_Varthur,location_Vidyaranyapura,location_Vijayanagar,location_Vittasandra,location_Whitefield,location_Yelachenahalli,location_Yelahanka,location_Yelahanka New Town,location_Yeshwanthpur,location_other
0,1672.0,3.0,150.00,3,8971.29,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
1,1750.0,3.0,149.00,3,8514.29,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
2,1750.0,3.0,150.00,3,8571.43,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
3,1250.0,2.0,44.00,3,3520.00,1,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
4,1250.0,2.0,40.00,2,3200.00,1,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
8912,2900.0,3.0,325.00,3,11206.90,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
8913,1780.0,3.0,84.83,3,4765.73,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
8914,880.0,2.0,48.00,2,5454.55,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
8915,1000.0,2.0,55.00,2,5500.00,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,1


In [9]:
df.to_csv(r"C:\Users\HP\Desktop\RealEstatePricePrediction\data\processed\featured_data.csv", index=False)
print("Feature engineered data saved!")
print("Final shape:", df.shape)

Feature engineered data saved!
Final shape: (8917, 181)
